torch_xla/csrc/ir.cpp

```diff
@@ -48,15 +48,30 @@ XlaNode::XlaNode(torch::lazy::OpKind op, torch::lazy::OpList operands,
     : torch::lazy::Node(op, operands, std::move(shapes), num_outputs),
       xla_shape_(std::move(xla_shape)),
       node_hash_(torch::lazy::HashCombine(op.hash(), hash_seed)),
-      dag_hash_(GetOperandHashes(operands, node_hash_)) {}
+      dag_hash_(GetOperandHashes(operands, node_hash_)),
+      oplist_(std::optional<torch::lazy::OpList>(operands)) {}
```

Comment:
Is it possible to construct OpList from `operands_` in `lazy::Node`? We are passing `operands` as input to the parent constructor and also to `oplist_`. Hopefully, that will free us from adding this additional constructor argument and possibly be safer, as we don't have to maintain two copies of `operands_` in two different formats. `operands_` is protected, so should be accessible.

Reply:
I followed your suggestion here to move the logic into XlaNode. Should I go back to using lazy::Node like I was originally? https://github.com/pytorch/pytorch/pull/92579/files#r1105075942

Reply:
No circular dependency here. Instead of doing it in `lazy::Node`, can we just take `operands_` from the parent (`lazy::Node`) and construct OpList in `XlaNode`? 

Also,

Reply:
Synced offline with @alanwaketan , `node(node)` from `torch::lazy::Value` ctor would hold the reference to `Node`.